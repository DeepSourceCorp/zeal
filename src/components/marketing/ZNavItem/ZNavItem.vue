<template>
  <z-button class="px-6 hover:bg-ink-300" :to="to" type="link">
    <slot></slot>
  </z-button>
</template>

<script lang="ts">
import Vue from 'vue'
import ZButton from '@/components/ZButton'

export default Vue.extend({
  components: { ZButton },
  name: 'ZNavItem',
  props: {
    to: {
      default: '',
      type: String
    }
  }
})
</script>
