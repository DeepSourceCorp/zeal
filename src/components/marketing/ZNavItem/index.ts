import ZNavItem from './ZNavItem.vue'

export default ZNavItem
