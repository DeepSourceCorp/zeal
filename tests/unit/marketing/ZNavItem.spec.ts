import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZNavItem from '../../../src/components/marketing/ZNavItem'

describe('NavItem component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZNavItem, {
        localVue,
        ...options
      })
    }
  })
  it('renders a ZNavItem when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })
})
