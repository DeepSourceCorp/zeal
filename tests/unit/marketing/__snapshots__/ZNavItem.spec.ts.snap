// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component renders a ZNavItem when the component is mounted 1`] = `<span class="text-vanilla-100"><!----> <span class="flex items-center cursor-pointer px-4 py-3 bg-ink-300 inline-block hover:opacity-75 select-none font-semibold w-full"><span class="flex flex-1"></span></span></span>`;
